compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -1115,19 +1141,15 @@ LogicalResult aie2xclbin(
   PassManager pm(ctx, mlir::ModuleOp::getOperationName());
   applyConfigToPassManager(pm, printIRBeforeAll, printIRAfterAll,
                            printIRModuleScope, timing);
-  // generateNPUInstructions
-  pm.addNestedPass<AIE::DeviceOp>(
-      mlir::iree_compiler::AMDAIE::createAMDAIEDmaToNpuPass());
   if (failed(pm.run(moduleOp)))
     return moduleOp.emitOpError(": NPU Instruction pipeline failed");
 
-  std::optional<ArrayRef<uint32_t>> npuInstructions =
-      cast<DenseUI32ResourceElementsAttr>(
-          (*moduleOp.getOps<xilinx::AIE::DeviceOp>().begin())
-              ->getAttr("npu_instructions"))
-          .tryGetAsArrayRef();
-  if (!npuInstructions)
-    return moduleOp.emitOpError(": No NPU instructions in device op");
+  auto maybeNpuInstructions = getNpuInstructions(moduleOp);
+  if (failed(maybeNpuInstructions)) {
+    assert(false && "Failed to get NPU instructions");
+    return failure();
+  }
+  auto npuInstructions = maybeNpuInstructions.value();
```

Comment:
here too

Reply:
In general MLIR conventions (and my own preference) is to not use auto unless it's a very verbose template-y kind of type.

Reply:
Not a fan of that convention, but will follow for code consistency here

Reply:
I mean the sword cuts both ways (too much typing vs. not enough typing) but one edge is much sharper I feel (having not enough typing and then swimming in a vast sea of `auto`) so the rule/style convention makes sense to me.

Reply:
A "vast sea of `auto`" sounds blissful to me. Maybe editor/plugin influences perceived sharpness of edges, if I want to know what type clang will choose I hover my cursor over `auto` (vim+ycm)

Reply:
you're overstimating ycm (which uses `clangd` under the hood for this) - it will absolutely fail in a sea of `auto` and you will just be completely _nonplussed_ about the type (how do I i know? mlir-aie's codebase was a sea `auto` before I scrubbed almost all of em out).